operator/internal/manifests/zone_aware_config.go

```diff
@@ -0,0 +1,52 @@
+package manifests
+
+import (
+	"strings"
+
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	availabilityZoneEnvVarName = "INSTANCE_AVAILABILITY_ZONE"
+	availabilityZoneFieldPath  = "metadata.annotations['" + lokiv1.AnnotationAvailabilityZone + "']"
+)
+
+var availabilityZoneEnvVar = corev1.EnvVar{
+	Name: availabilityZoneEnvVarName,
+	ValueFrom: &corev1.EnvVarSource{
+		FieldRef: &corev1.ObjectFieldSelector{
+			FieldPath: availabilityZoneFieldPath,
+		},
+	},
+}
+
+func configureReplication(podTemplate *corev1.PodTemplateSpec, replication *lokiv1.ReplicationSpec, component string, stackName string) {
```

Comment:
nit: added this comment to allow people to more easily understand what is the goal of this function
```suggestion
// configureReplication is responsible for setting up the TopologySpreadContraints of the podTemplate and adding
// the metadata (LabelZoneAwarePod and AnnotationAvailabilityZoneLabels) required by the
// ZoneAwarenessController to enable zone awareness behaviour.
func configureReplication(podTemplate *corev1.PodTemplateSpec, replication *lokiv1.ReplicationSpec, component string, stackName string) {
```

Reply:
WDYT of adding a unit test for this function?

Reply:
Each component tests this function in tests similar to this - https://github.com/grafana/loki/pull/9503/files#diff-ead163e4b6e1d91440ff4f5bc70a1d6050832a328d299bff703cf1459766581dR110